fiftyone/core/labels.py

```diff
@@ -561,7 +561,7 @@ def to_segmentation(self, mask=None, frame_size=None, target=255):
 
         mask, _ = _parse_to_segmentation_inputs(mask, frame_size, None)
         _render_instance(mask, self, target)
-        return Segmentation(mask=mask)
+        return Segmentation(mask=mask.astype(np.int))
```

Comment:
`_parse_to_segmentation_inputs()` creates a `uint8` mask by default, so if we need to use `int`, then it would be best to specify it there rather than do a cast in all of these return statements.

Is there a particular reason you need `int` as opposed to `uint8`? 

In fact, now that I think of it, this change probably breaks things like [segmentation exports](https://github.com/voxel51/fiftyone/blob/5063f248a247ffc4a8f56d2d19201d3a190527e4/fiftyone/utils/data/exporters.py#L2398-L2431), which need to write masks as images, which, unless we do some work, will require 8 bit grayscale.

I'm struggling a bit to decide what dtypes to use/allow for semantic segmentations. Whenever FiftyOne internal code creates masks, I've tended to use `uint8` to save space and for export reasons mentioned above, while [the docs](https://voxel51.com/docs/fiftyone/user_guide/using_datasets.html#semantic-segmentation) say "any integer type", and the code itself doesn't do any validation at all...

On the one hand, the smallest possible dtype for arrays stored in the database is a pretty big deal, as `uint8` (8 bits) is 4x-8x smaller and thus faster to work with than `int` (defaults to 32 bits, unless a large enough value is inserted, in which case it silently upgrades to 64 bits, I believe).

On the other hand, this does implicitly limit to 255 classes + background, which might be too few for some users (but this would be rare, I think, given that segmentations are often stored on disk as images).

But, of course, 8 bits only allows for 256 classes

Reply:
Right, I really should have pointed this out specifically. The reason I did that was because the App wasn't showing masks stored in `uint8`, only when stored in `int` so I assumed that when the docs said "any integer type", it meant `int`. I agree that `uint8` should be the default and it may have just been an App issue. Have you encountered that?

Reply:
Oh I see. That was reported in https://github.com/voxel51/fiftyone/issues/1204 but Ben fixed it in https://github.com/voxel51/fiftyone/pull/1205 a few weeks ago

Reply:
Since the App should now any integer type, I'll take a whack at improving the export logic so that non-uint8 segmentations can be appropriately exported (I guess as 16 bit images

If the cast isn't need here anymore, then I think we should remove it and revisit the issue of segmentation mask typing separately.

Reply:
There it is! I didn't rebuild my development App since that update apparently. Yeah, this change should be removed then.